activerecord/lib/active_record/reflection.rb

```diff
@@ -708,10 +708,13 @@ def inverse_name
         def automatic_inverse_of
           if can_find_inverse_of_automatically?(self)
             inverse_name = ActiveSupport::Inflector.underscore(options[:as] || active_record.name.demodulize).to_sym
-            plural_inverse_name = ActiveSupport::Inflector.pluralize(inverse_name)
 
             begin
-              reflection = klass._reflect_on_association(inverse_name) || klass._reflect_on_association(plural_inverse_name)
+              reflection = klass._reflect_on_association(inverse_name)
+              if !reflection && klass.automatic_scope_inversing == :plurals
+                plural_inverse_name = ActiveSupport::Inflector.pluralize(inverse_name)
+                reflection = klass._reflect_on_association(plural_inverse_name)
+              end
```

Comment:
Prior to this PR, I'd explored this as a deprecation warning instead. Is it important to add `options[:inverse_of].nil?` to this condition?

Reply:
> Is it important to add `options[:inverse_of].nil?` to this condition?

Not sure I follow. If you pass `inverse_of: nil` we already don't enter this code path.

Reply:
My mistake. What I mean is `!options.key?(:inverse_of)`. If you _aren't_ being explicit about the `:inverse_of` option. If you pass any option, AR should try to use it.